plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/UriTemplateServerProxyTransformer.java

```diff
@@ -64,21 +70,46 @@ public ServerImpl transform(ServerImpl server) {
         if (serverPath.startsWith("/")) {
             serverPath = serverPath.substring(1);
         }
-
+        final String externalAddress;
+        final String externalServerHost;
+        final String internalServerHost;
+        if (!Strings.isNullOrEmpty(externalAddressProperty) && !Strings.isNullOrEmpty(codenvyHost)) {
```

Comment:
But in dockerized Codenvy it is always set? If so will multiple nodes of swarm work?